### DISCUSSION TITLE
Named Steps

### DISCUSSION THREAD

**Discussion by luixal** [mealie-recipes/mealie#4242] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Steps named "Step 1" are not really useful in the fronted, i.e:

![image](https://github.com/user-attachments/assets/dad14ca7-eb94-4c97-9f87-ff7c19a11f70)

renaming the step would allow to get something like this, where a collapse recipe makes much more sense:

![image](https://github.com/user-attachments/assets/5a4964b3-3f11-4b2d-a699-408df4238c07)

you can even get rid of sections (they're nice though) if you could add a background color to the step's title:

![image](https://github.com/user-attachments/assets/2b8dd906-40aa-462c-96a0-844899f7f751)

I tend to make my recipes this way as that means less steps as you can reduce all the verbose stuff in one step you need to read altogether.

### Please provide a concise description of the feature that would resolve your issue.

I would like to be able to rename a step right in the repice editor.

### Please consider and list out some caveats or tradeoffs made in your design decision

None. 

In the case I wanted to still have an index, I can include it in the name.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
This has been iplemented by #4410
